### PR TITLE
"Implemented" `MainMenu::ReplayHandling`

### PR DESCRIPTION
**Pull request by PieVieRo**
it works, not counting some minor mistakes. also it differs quite a bit from the assembly

### CODE REVIEW

File: src/MainMenu.cpp [hunk 1]
```diff
@@ -311,3 +315,235 @@ ZunResult MainMenu::DrawStartMenu(void)
...
+                        // FIXME: wrong assembly
+                        memcpy(&this->replayFileData[replayFileIdx], replayData, 0x50);
+                        // HACK: i dont think it should be this way
+                        _strcpy(this->replayFilePaths[replayFileIdx], replayFilePath);
+                        sprintf(this->replayFileName[replayFileIdx], "No.%.2d", cur + 1);
+                        replayFileIdx++;
+                    }
+                    free(replayData);
+                }
+                FileSystem::CreateDirectoryInCWD("./replay");
```
Comment (roblabla) [inline src/MainMenu.cpp:371]: This is almost certainly a function from the C standard library, such as [`_mkdir`](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/mkdir-wmkdir?view=msvc-170).

File: src/MainMenu.cpp [hunk 2]
```diff
@@ -311,3 +315,235 @@ ZunResult MainMenu::DrawStartMenu(void)
     return ZUN_SUCCESS;
 }
 #pragma optimize("", on)
+
+void _strcpy(char *dst, char *src)
```
[the remark this answers is not in the document]
Comment (PieVieRo) [reply src/MainMenu.cpp:319]: when i do it without the alias it generates different assembly for some reason

[the remark this answers is not in the document]
Comment (roblabla) [reply src/MainMenu.cpp:320]: Ah, it's probably generating an intrinsic? You can disable the intrinsic using:

```
#pragma function(strcpy)
```

Before the function.

File: src/MainMenu.cpp [hunk 3]
```diff
@@ -311,3 +316,235 @@ ZunResult MainMenu::DrawStartMenu(void)
...
+                    anmVm->pendingInterrupt = 0x11;
+                }
+                anmVm = &this->vm[99 + this->chosenReplay];
+                anmVm->pendingInterrupt = 0x10;
+                this->stateTimer = 0;
+                this->cursor = 0;
+                g_SoundPlayer.PlaySoundByIdx(10, 0);
+                this->currentReplay = (ReplayData *)FileSystem::OpenPath(this->replayFilePaths[this->chosenReplay], 1);
+                ValidateReplayData(this->currentReplay, g_LastFileSize);
+                for (cur = 0; cur < ARRAY_SIZE_SIGNED(this->currentReplay->stageScore); cur++)
```
[the remark this answers is not in the document]
Comment (PieVieRo) [reply src/MainMenu.cpp:445]: yeah i copypasted the wrong commit back when pulling deleted everything

its correct now in the latest commit

File: src/MainMenu.cpp [hunk 4]
```diff
@@ -311,3 +316,235 @@ ZunResult MainMenu::DrawStartMenu(void)
...
+                    {
+                        this->currentReplay->stageScore[cur] =
+                            (StageReplayData *)((u32)this->currentReplay + (u32)this->currentReplay->stageScore[cur]);
+                    }
+                }
+
+                do
+                {
+                    // FIXME: there's an additional jump
+                    if (!this->replayFileData[this->chosenReplay].stageScore[this->cursor])
```
Comment (roblabla) [inline src/MainMenu.cpp:457]:
```suggestion
                    if (this->replayFileData[this->chosenReplay].stageScore[this->cursor])
```

File: src/MainMenu.cpp [hunk 5]
```diff
@@ -311,3 +316,235 @@ ZunResult MainMenu::DrawStartMenu(void)
...
+                    if (this->currentReplay->stageScore[cur] != NULL)
+                    {
+                        this->currentReplay->stageScore[cur] =
+                            (StageReplayData *)((u32)this->currentReplay + (u32)this->currentReplay->stageScore[cur]);
+                    }
+                }
+
+                do
+                {
+                    // FIXME: there's an additional jump
```
Comment (roblabla) [inline src/MainMenu.cpp:456]: My gut feeling is that the whole do/while/goto/return is slightly wrong somehow, but I'm having a hard time figuring out what the proper control flow should be. My gut feeling is that the `goto` should be a break, and the return might be gated behind an if or something?

**Pull request merged**